pkg/reconciler/revision/resources/pa_test.go

```diff
@@ -308,6 +310,56 @@ func TestMakePA(t *testing.T) {
 				// Reachability trumps failure of Revisions.
 				Reachability: autoscalingv1alpha1.ReachabilityUnknown,
 			}},
+	}, {
+		name: "name is bruce (Revision pending)",
+		rev: func() *v1.Revision {
+			rev := v1.Revision{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "blah",
+					Name:      "bruce",
+					UID:       "4321",
+				},
+				Status: v1.RevisionStatus{
+					Status: duckv1.Status{
+						Conditions: duckv1.Conditions{{
+							Type:   v1.RevisionConditionReady,
+							Status: "False",
+						}},
+					},
+				},
+			}
```

Comment:
```suggestion
			}
			rev.Status.MarkResourcesAvailableFalse("foo", "bar")
```

Reply:
The thing is there are two conditions that lead to the same `Unreachable` status, so I did more testing with the e2e test that was described in the issue, not sure what's the best approach here

Reply:
cc @dprotaso for input too

Reply:
can you a look @dprotaso